_posts/2022-08-05-hacking-for-cruise-control-ui.md

```diff
@@ -0,0 +1,279 @@
+# cruise-control-ui.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cruise-control-ui
+  labels:
+    app: cruise-control-ui
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: cruise-control-ui
+  template:
+    metadata:
+      labels:
+        app: cruise-control-ui
+    spec:
+      containers:
+      - name: cruise-control-ui
+        image: <registry>/<cruise-control-ui-image-name>
+        ports:
+        - containerPort: 80
+        volumeMounts:
+        - name: cruise-control-ui
+          mountPath: /etc/nginx/conf.d
+      volumes:
+        - name: cruise-control-ui
+          configMap:
+            name: cruise-control-ui
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cruise-control-ui
+  labels:
+    app: cruise-control-ui
+data:
+  default.conf: |2+
+        server {
+        listen       80;
+        listen  [::]:80;
+        server_name  localhost;
+
+        location / {
+            root   /usr/share/nginx/html;
+            index  index.html index.htm;
+        }
+
+        location /kafkacruisecontrol/ {
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_buffering off;
+            proxy_pass http://my-cluster-cruise-control:9090/kafkacruisecontrol/;
+        }
+        
+        error_page   500 502 503 504  /50x.html;
+        location = /50x.html {
+            root   /usr/share/nginx/html;
+        }
+        }
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: cruise-control-ui
+  labels:
+    app: cruise-control-ui
+spec:
+  selector:
+    app: cruise-control-ui
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 80
+```
+
+Then run it:
+```
+kubectl apply -f cruise-control-ui.yaml -n <namespace>
+```
+
+## Using Cruise Control Front End
+
+After being deployed, the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) application will be accessible via the cruise-control-ui `service`.
+
+If running the cluster locally with [minikube](https://github.com/kubernetes/minikube), you can port-forward the Cruise Control UI `service` to access it locally:
+
+```
+kubectl port-forward svc/cruise-control-ui 8090:80
+```
+
+Then navigate [‘http://127.0.0.1:8090’](http://127.0.0.1:8090) in your browser
+
+![Cruise Control Frontend](/assets/images/posts/2019-08-08-cruise-control-frontend.png)
+
+## Conclusion
+
+Although it is possible to run a [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) UI instance alongside a Strimzi-managed Kafka cluster it is only possible when the Cruise Control REST API security settings are disabled.
+Having these security settings disabled exposes the Cruise Control API to potentially destructive Cruise Control operations that are not coordinated or supported by Strimzi Operator.
+If using the Cruise Control UI with a Strimzi-managed Kafka cluster, it is best to stick with using it solely for monitoring your cluster, not for operating on it.
```

Comment:
It's a bit disappointing that these caveats only come at the end, after you've got people excited about this UI thing that looks somewhat like the ever-missing Strimzi UI. So maybe we should have a word of caution at the start too so people understand from the start that the UI is only _really_ safe for read-only interactions.